pydantic/dataclasses.py

```diff
@@ -103,7 +103,11 @@ def _process_class(
     def _pydantic_post_init(self: 'Dataclass', *initvars: Any) -> None:
         if post_init_original is not None:
             post_init_original(self, *initvars)
-        d, _, validation_error = validate_model(self.__pydantic_model__, self.__dict__, cls=self.__class__)
+
+        # We need to remove `FieldInfo` values since they are not valid as input
+        # It's ok since they are obviously the default values
+        input_data = {k: v for k, v in self.__dict__.items() if not isinstance(v, FieldInfo)}
```

Comment:
Humm, I wonder if this will have an adverse effect on performance?

Perhaps we could add a flag somehow below when any fields use `Field`, then only run this logic if so?

Reply:
relevant commit fb97e08